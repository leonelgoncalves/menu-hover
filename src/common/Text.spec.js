import React from "react";
import { mount, shallow } from "enzyme";
import TextComponent from "./Text";
import renderer from "react-test-renderer";

describe("Text:", () => {
  let props;
  let mountedText;
  const text = () => {
    if (!mountedText) {
      mountedText = mount(<TextComponent {...props} />);
    }
    return mountedText;
  };

  beforeEach(() => {
    props = {
      text: undefined,
      textClass: undefined,
      textColor: undefined,
      textRef: undefined
    };
    mountedText = undefined;
  });

  it("always render span", () => {
    const spans = text().find("span");
    expect(spans.length).toBeGreaterThan(0);
  });
});

describe("Text Snapshot:", () => {
  it("should renders correctly", () => {
    expect(shallow(<TextComponent text="Hola!" />)).toMatchSnapshot();
  });

  it("should add color to text", () => {
    expect(
      shallow(<TextComponent textColor="red" text="Hola!" />)
    ).toMatchSnapshot();
  });
});
