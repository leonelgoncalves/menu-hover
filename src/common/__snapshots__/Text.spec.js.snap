// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text Snapshot: should add color to text 1`] = `
<span
  className=""
  style={
    Object {
      "color": "red",
    }
  }
>
  Hola!
</span>
`;

exports[`Text Snapshot: should renders correctly 1`] = `
<span
  className=""
  style={
    Object {
      "color": "",
    }
  }
>
  Hola!
</span>
`;
